### ISSUE TITLE
Tracking Install Referrer

### ISSUE THREAD

**Issue by mateuszgwozdz** [arnesson/cordova-plugin-firebase#809]
I cannot find any information about install referrer. How can I track it?

**Comment by briantq** [comment]
I am going to argue this should not be part of the this plugin.  The Install Referrer is not part of Firebase.  In fact, it requires simply an Android dependency.  

Firebase itself is a large set of functionality and if we try to take on even more scope this plugin could become more of a maintenance nightmare.  Additionally, it could interfere with other users who have implemented it using recommended routes, like using this plugin https://github.com/vijayrajasekaran/cordova-plugin-referrer (found via a stack overflow answer). 

@mateuszgwozdz @guylando I'll leave it open for additional discussion.  Please let me know why you think it _should_ be part of this plugin

**Comment by briantq** [comment]
@guylando you are correct, this plugin is intended to expose Firebase APIs.  From what I found, the Install Referrer functionality comes from Google Play Services, https://developer.android.com/google/play/installreferrer/library.

Are we talking about something different?

**Comment by guylando** [comment]
@briantq 
1. Well, underneath probably firebase itself uses some of google play services too but exposes it in cross platform way supporting ios too.
2. The link indeed speaks about the most low level way of getting the install referrer on android devices. For example appsflyer gets it this way or another way on both android and iphone devices.
3. The question is if firebase has some functionality related to this. As you can see here: https://support.google.com/firebase/answer/6317485?hl=en firebase DOES already integrate with the relevant event of first_open. So it means firebase already has a cross platform way of getting the first_open event, so I would expect the install referrer to be part of it.

**Comment by briantq** [comment]
@guylando Thanks for the article link.  My interpretation is that Firebase internally processes different events based on "As long as you use the Firebase SDK, you don't need to write any additional code to collect these events".  

Additionally, looking at the [Firebase Documentation](https://firebase.google.com/docs/analytics/android/start/), you can find information on setting/adding events, but as far as I can tell there is no API to retrieve event information.  Please let me know if you see something else.

So, if you can achieve a solution to get the install referrer without hacking Firebase or adding additional plugin dependencies (like Google Play Services), I think that would make a good candidate.  Does that make sense?

**Comment by kaoecoito** [comment]
To implement the install referrer trace I used firebase dynamic links, which I saw is the simplest and most efficient way to implement.
Since this plugin does not support this feature I used the firebase-dynamic-links plugin, however I had to make a fork and modify some things in the plugin because it conflicts with this firebase plugin, and I need both in my application.
I understand the comment about having separate plugins for each functionality of the firebase but I see the plugins in the cordova generate a lot of conflict, especially when using google services.
I believe it is more functional to have all the functions of the firebase in a single plugin to avoid the conflicts, since in general one does not use only isolated firebase functionality. The other option is to create some basic firebase deployments project to later have individual, non-conflicting plugins.
